addons/l10n_in/models/res_partner.py

```diff
@@ -98,5 +98,6 @@ def check_vat_in(self, vat):
 
     def action_update_state_as_per_gstin(self):
         self.ensure_one()
-        state_id = self.env['res.country.state'].search([('l10n_in_tin', '=', self.vat[:2])], limit=1)
-        self.state_id = state_id
+        if self.vat and self.check_vat_in(self.vat):
```

Comment:
The existence a vat is already verified in `check_vat_in` so no need to do it twice but other than that the fix seems fine :+1: 
 
```suggestion
        if self.check_vat_in(self.vat):
```